api/nodes/views.py

```diff
@@ -228,12 +294,62 @@ def get_default_odm_query(self):
         query = base_query & permission_query
         return query
 
-    # overrides ListCreateAPIView
+    # overrides ListBulkCreateUpdateDestroyAPIView
     def get_queryset(self):
-        query = self.get_query_from_request()
-        return Node.find(query)
+        # For bulk requests, queryset is formed from request body.
+        if isinstance(self.request.data, list):
+            query = Q('_id', 'in', [node['id'] for node in self.request.data])
+
+            user = self.request.user
+            if user.is_anonymous():
+                auth = Auth(None)
+            else:
+                auth = Auth(user)
 
-    # overrides ListCreateAPIView
+            nodes = Node.find(query)
+            for node in nodes:
+                if not node.can_edit(auth):
```

Comment:
Generic views don't apply object level permissions to every instance in the queryset when returning a list of objects; you filter them instead.  Bulk update of nodes is using the same view as GET /v2/nodes/.  Since forbidden nodes are filtered out in get_queryset for GET, I thought I could do it in the same place for bulk.